src/app/children/previous-schools/previous-schools.component.ts

```diff
@@ -0,0 +1,88 @@
+import {Component, OnInit, ViewChild} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {DatePipe} from '@angular/common';
+import {ChildSchoolRelation} from '../childSchoolRelation';
+import {ColumnDescription} from '../../ui-helper/entity-subrecord/column-description';
+import {ChildrenService} from '../children.service';
+import {SchoolsService} from '../../schools/schools.service';
+import { School } from 'app/schools/school';
+
+@Component({
+  selector: 'app-previous-schools',
+  template: '<app-entity-subrecord [records]="records" [columns]="columns" [newRecordFactory]="newRecordFactory" [optionalFormValidation]="optionalFormValidation">' + '</app-entity-subrecord>',
+})
+
+
+export class PreviousSchoolsComponent implements OnInit {
+
+  childId: string;
+  records = new Array<ChildSchoolRelation>();
+  schoolList = new Array<School>();
+  columns = new Array<ColumnDescription>();
+
+  constructor(private route: ActivatedRoute,
+              private childrenService: ChildrenService,
+              private schoolsService: SchoolsService,
+              private datePipe: DatePipe) {
+  }
+
+
+  ngOnInit() {
+    this.route.paramMap.subscribe(params => {
+      this.childId = params.get('id').toString();
+      this.loadData(this.childId);
+    });
+  }
+
+
+  loadData(id: string) {
+    let tempRecords = new Array<ChildSchoolRelation>();
+    this.childrenService.getSchoolsWithRelations(id)
+      .then(results => {
+        tempRecords = results;
+        this.records = tempRecords
+          .sort((a, b) => (
```

Comment:
The sorting should already be performed by the used indices and therefore can be removed here